langchain4j/src/main/java/dev/langchain4j/agent/tool/DefaultToolExecutor.java

```diff
@@ -109,14 +108,52 @@ static Object[] prepareArguments(
             if (argumentsMap.containsKey(parameterName)) {
                 Object argument = argumentsMap.get(parameterName);
                 Class<?> parameterType = parameters[i].getType();
-
-                arguments[i] = coerceArgument(argument, parameterName, parameterType);
+                if( Collection.class.isAssignableFrom(parameterType) || Map.class.isAssignableFrom(parameterType))  {
```

Comment:
nit: indentations after `if` are different in the same file. Maybe we can align them to be the same to follow similar codestyle?

Reply:
Did get it. It looks good only.